packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -670,6 +800,61 @@ export class IntervalCollectionView<TInterval extends ISerializableInterval> ext
     }
 }
 
+export class IntervalCollectionIterator<TInterval extends ISerializableInterval> {
+    private readonly results: TInterval[];
+    private index: number;
+
+    constructor(
+        collection: IntervalCollection<TInterval>,
+        iteratesBackward: boolean = false,
+        hasStartPosition: boolean = false,
```

Comment:
more of a question. it seems like we have a lot of overloads, do we need to keep start and end position so seperate? could we just have gather forward and backward, and they take a start and end, both optional, so something like pos: {start?:number,end?:number}

Reply:
Sounds good. I'm reducing the number of overloads in the next version.